sdk/nodejs/src/models/populatedPBUser.ts

```diff
@@ -1,84 +1,50 @@
-import { DVCCustomDataJSON } from '../types'
-import * as packageJson from '../../package.json'
-import { DevCycleUser } from './user'
-import os from 'os'
 import { ProtobufTypes } from '@devcycle/bucketing-assembly-script'
+import {
+    DVCPopulatedUser,
+    DevCycleUser,
+    DVCCustomDataJSON,
+} from '@devcycle/js-cloud-server-sdk'
+import { getNodeJSPlatformDetails } from '../utils/platformDetails'
 
-export class DVCPopulatedUser implements DevCycleUser {
-    user_id: string
-    email?: string
-    name?: string
-    language?: string
-    country?: string
-    appVersion?: string
-    appBuild?: number
-    customData?: DVCCustomDataJSON
-    privateCustomData?: DVCCustomDataJSON
-    readonly lastSeenDate: Date
-    readonly createdDate: Date
-    readonly platform: string
-    readonly platformVersion: string
-    readonly sdkType: 'server'
-    readonly sdkVersion: string
-    readonly hostname: string
-
+export class DVCPopulatedPBUser extends DVCPopulatedUser {
     constructor(user: DevCycleUser) {
-        this.user_id = user.user_id
-        this.email = user.email
-        this.name = user.name
-        this.language = user.language
-        this.country = user.country
-        this.appVersion = user.appVersion
-        this.appBuild = user.appBuild
-        this.customData = user.customData
-        this.privateCustomData = user.privateCustomData
-        this.lastSeenDate = new Date()
-
-        /**
-         * Read only properties initialized once
-         */
-        this.createdDate = new Date()
-        this.platform = 'NodeJS'
-        this.platformVersion = process.version
-        this.sdkType = 'server'
-        this.sdkVersion = packageJson.version
-        this.hostname = os.hostname()
+        super(user, getNodeJSPlatformDetails())
     }
 
     toPBUser(): ProtobufTypes.DVCUser_PB {
         const params = {
-            user_id: this.user_id,
+            user_id: super.user_id,
```

Comment:
why super and not this?

Reply:
params are defined on the extended class, so `this` didn't work, but `super` did.